magpie/register.py

```diff
@@ -607,11 +605,9 @@ def _log_permission(message, permission_index, trail=", skipping...", detail=Non
     .. seealso::
         `magpie/config/permissions.cfg`
     """
-    if detail:
-        trail = "{}\nDetail: [{!s}]".format(trail, detail)
-    if permission:
-        permission = " [{!s}]"
-    LOGGER.log(level, "{!s} [permission #{}]{}{}".format(message, permission_index, permission or "", trail or ""))
+    trail = "{}\nDetail: [{!s}]".format(trail, detail) if detail else (trail or "")
+    permission = " [{!s}]".format(permission) if permission else ""
+    LOGGER.log(level, "{!s} [permission #{}]{}{}".format(message, permission_index, permission, trail))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use % formatting in logging functions and pass the % parameters as arguments](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650806)